.github/workflows/test.yml

```
@@ -35,19 +35,27 @@ jobs:
     - run: npm test
   tox:
     name: >-
-      ${{ matrix.env.TOXENV }}
+      ${{ matrix.toxenv }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        python-version:
-        - 3.8
-        os:
-        - ubuntu-20.04
-        env:
-        - TOXENV: py,lint
+        include:
+        - python-version: 3.8
+          os: ubuntu-20.04
+          toxenv: py38
+        - python-version: 3.9
+          os: ubuntu-20.04
+          toxenv: py39
+        - python-version: '3.10'
+          os: ubuntu-20.04
+          toxenv: py310
+        - python-version: 3.8
+          os: ubuntu-20.04
+          toxenv: lint
     env:
       TOX_PARALLEL_NO_SPINNER: 1
+      TOXENV: ${{ matrix.toxenv }}
```

It is also possible to do

```suggestion
      TOXENV: ${{ matrix.noxenv && matrix.toxenv || 'py' }}
```

And drop the explicit toxenv entries from the job definitions except for lint.